modules/bismark/methylationextractor/main.nf

```diff
@@ -31,6 +31,13 @@ process BISMARK_METHYLATIONEXTRACTOR {
     path "versions.yml"                            , emit: versions
 
     script:
+    // Assign sensible numbers for multicore and buffer_size based on bismark docs
+    def ccore = task.cpus ? ((task.cpus as int) / 3) as int : 1
+    def multicore = (ccore > 1) ? "--multicore ${ccore}" : ""
+    // Only set buffer_size when there are more than 6.GB of memory available
+    def mbuffer = task.memory ? (task.memory as nextflow.util.MemoryUnit) - 2.GB : (4.GB).toBytes()
+    def buffer = (mbuffer.compareTo(4.GB) == 1) ? "--buffer_size ${mbuffer.toGiga()}G" : ""
```

Comment:
```suggestion
    def buffer = (task.memory.giga >= 6) ? "--buffer_size ${task.memory.giga - 2}G" : ""
```

Do we really need to check if `task.memory` is TRUEish? Are there cases where it can be `null`?
Unless I'm missing something obvious, this code could be simplified further.

Reply:
Yeah it can be `null`. Didn't use to have this until someone came along with a cluster that doesn't allow job submissions asking for memory (only cpus).

Reply:
I see..
This should work though (safe navigation):

```suggestion
    def buffer = (task.memory?.giga > 6) ? "--buffer_size ${task.memory.giga - 2}G" : ""
```

Reply:
Nope, some configs [explicitly overwrite to give `null`](https://github.com/nf-core/configs/blob/0866682e9b38acf8cc56b087ed21eb2ccb841cfb/conf/hebbe.config#L16-L17) as it's the only way to get the pipeline to run with their job scheduler.

But yeah, your suggestion is more succinct 🙂